javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java

```diff
@@ -36,6 +39,11 @@ public static synchronized void installAgentTracer() {
       OpenTelemetrySdk sdk = OpenTelemetrySdkAutoConfiguration.initialize();
       OpenTelemetrySdkAccess.internalSetForceFlush(
           (timeout, unit) -> sdk.getSdkTracerProvider().forceFlush().join(timeout, unit));
+
+      if (Config.get().isInstrumentationEnabled(Collections.singleton("runtime-metrics"), true)) {
+        GarbageCollector.registerObservers();
+        MemoryPools.registerObservers();
+      }
```

Comment:
What do you think about creating a separate `ComponentInstaller` in the `:instrumentation:runtime-metrics:javaagent` module? This way we could avoid adding instrumentation code (well, sort of) to javaagent-tooling module.

Reply:
^^ This is a VERY good idea

Reply:
oh, that is a good idea 👍

Reply:
Hmm - I tried and it's not trivial since it leads to a dependency cycle in Gradle, `javaagent-tooling` depends on `runtime-metrics:javaagent` depends on `javaagent-tooling`. I'm also not sure it's that much better - it's nowhere close to the pattern we use in any other `javaagent` module so working around these issues seems to be counter-intuitive?

Reply:
BTW, I think the way I'd solve it is to have a folder, ideally not called `javaagent` since it seems so inconsistent, that does not `apply from: instrumentation.gradle` since it doesn't need any of its patterns. And have a `javaagent-testing` folder which does which only includes unit tests. It's highly inconsistent though.

Reply:
`ComponentInstaller` is in `javaagent-spi` not `javaagent-tooling`, does using that directly break the cycle?

Reply:
We add javaagent-tooling to all the instrumentations. But we've refactored a lot, perhaps that's not needed anymore let me see if there's a simple fix

Reply:
oh right, all the bytebuddy matchers are in javaagent-tooling, I'm good with this as-is, and we can revisit later how to decouple it

Reply:
Hmm, why does `javaagent-tooling` need to depend on `runtime-metrics:javaagent`? It does not depend directly on all other instrumentations, they all get added to `javaagent` directly.

Reply:
@mateuszrzeszutek Oops - forgot it'd be automatically added, yeah let me try it